Assets/MRTK/SDK/Features/UX/Scripts/Pointers/SpherePointer.cs

```diff
@@ -598,8 +595,12 @@ public bool IsColliderValidGrabbable(Collider collider, bool ignoreCollidersNotI
 
                 if (!isValidGrabbable)
                 {
-                    // Remove it from the cache if the grabbable is no longer valid for the object
-                    componentCache.Remove(instanceId);
+                    if (currentGrabbable == null)
```

Comment:
I think the null check is overly restrictive here. The currentGrabbable could potentially have become a boundsHandles through some outside means, or the sphere pointer could have had the `ignoreBoundsHandlesForQuery` flag changed suddenly. In any case, there is an edge case where we want to remove the grabbable from the cache, even though it is not null.

Reply:
> In any case, there is an edge case where we want to remove the grabbable from the cache, even though it is not null.

This is exactly the scenario I'm avoiding. I don't agree that we want to remove the grabbable from the cache if it's invalid, because then we're losing the main benefit of the cache!

If we remove the grabbable from the cache just because it's invalid, the next time through this logic we'll be calling `GetComponent` _again_, just to add and immediately remove the invalid component from the cache.

With this change, we now keep _all_ grabbables in the cache until they become null or until they "age out" of the LRUCache, to more fully take advantage of the "use an LRUCache to prevent repeated `GetComponent` calls" pattern. This method will still return `false` like it used to, so the grabbable will be ignored whether it's in the cache or not.

Reply:
Ok that makes sense, can you add a small description of why we only remove it from the cache in the null case? Something like "Only remove from the cache when the value is null to avoid future GetComponent calls" or something.

Reply:
Added a longer comment here for clarity